code/ui/blocks/src/components/DocsPage.tsx

```diff
@@ -1,9 +1,32 @@
 import { withReset } from '@storybook/components';
+import type { CSSObject } from '@storybook/theming';
 import { styled } from '@storybook/theming';
 import { transparentize } from 'polished';
 import type { FC } from 'react';
 import React from 'react';
 
+/**
+ * This selector styles all raw elements inside the DocsPage like this example with a `<div/>`:
+ * :where(div:not(.sb-unstyled, .sbdocs-preview... :where(.sb-unstyled, .sbdocs-preview...) div))
+ *
+ * 1. first ':where': ensures this has a specificity of 0, making it easy to override.
+ * 2. 'div:not(...)': selects all div elements that are not...
+ * 3. '.sb-unstyled, .sbdocs-preview...': any of the elements we don't want to style
+ * 3. ':where(.sb-unstyled, .sbdocs-preview...) div': or are descendants of an .sb-unstyled or .sbdocs-preview, etc. It is a shorthand for '.sb-unstyled div, sbdocs-preview div...'
+ * 4. .sb-unstyled is an escape hatch that allows the user to opt-out of the default styles
+ *    by wrapping their content in an element with this class: <Unstyled />
+ * 5. the other UNSTYLED_SELECTORS are elements we don't want the styles to bleed into, like canvas, story and source blocks.
+ */
+const UNSTYLED_SELECTORS = [
+  '.sb-unstyled',
+  '.sbdocs-preview',
+  '.sb-story',
+  '.docblock-source',
+  '.sb-anchor',
```

Comment:
It's up for discussion how the responsibility should be divided here.

This selector could also just be `sb-unstyled`, and then it was the responsibility of any of the components (`Story`, `Canvas`, etc.) to add that class to their elements.

Who is responsible for unstyling elements, the styler (here) or the element?

Reply:
I'm open to whatever but my instinct was the other way round (`.sb-unstyled`)

Reply:
Let's keep it as-is then and revisit if we get wiser.

Reply:
Narrator: He did get wiser. https://github.com/storybookjs/storybook/pull/21150